storage/oasis/nodeapi/api.go

```diff
@@ -206,10 +206,54 @@ type Proposal governance.Proposal
 // ....................  Runtimes  ....................
 // ....................................................
 
+// FallibleResponse stores the response to a node query. If the call was successful,
+// the response is stored in `result`. If the call failed due to a deterministic
+// error, eg call reverted, the error is stored in `deterministicErr`. This struct
+// should _not_ store transient errors or any other errors not considered to be a
+// valid node response.
+type FallibleResponse[Response any] struct {
```

Comment:
I suggest skipping the generics until we need them. It will keep things simpler. We can also easily add them back in if ever needed.

Reply:
good point - removed